feature-selection/feature-selection-main.metta

```diff
@@ -0,0 +1,71 @@
+;; the main feature selector procedure
+;;      preprocess the exemplar 
+;;                          -- gets column indices of features used in the exemplar
+;;                          -- checks two parameteres
+;;                                  1. increase target size of the feature selection by number of features in the examplar -- or just use exemplar features as seed
+;;                                  2. or ignore the features in exemplar in the feature selection process
+;;                          $targetSize -- the number of features sets to be returned
+;;                          $increaseTargetSize -- bool -- to increase number of features in a feature set by the number of features already in the exemplar
+
+(: featureSelector (->  Symbol (ITable $a) Number (Tree $a) Number Number Bool Expression Bool Number Bool Symbol Number Expression))
+(= (featureSelector 
+            $algo               ;; feature selection algo one of the following symbols hc, rnd, sim, inc, smd for hillclimbing, random, simple, incremental and stochasticMaxDependency selection algos
+            (mkITable $table $labels)   ;; the table
+            $th                 ;; score threshold (MI for instance)
+            $exemplar           ;; the exemplar tree
+            $diversity-cap      ;; cap on population of feature sets
+            $target-size        ;; number of features in a feature set
+            $init-xmplr-fts     ;; bool to determine to use exemplar features as initial features or not
+            $init-fts           ;; initial features specified list of symbols -- column labels
+            $ignore-xmplr-fts   ;; bool dictating whether to/not ignore exemplar features in feature selection
+            $div-pres           ;; diversity pressure -- XXX more explanation 
+            $multi-diversity    ;; multiply diversity -- used when updating the penalized score
+            $sim-scorer         ;; feature set similarity scorer either mi or ja for jaccard
+            $nDeme              ;; number of demes to spawn
+)
+        (let* (($exemplar-fts (treeFtsIndices $exemplar $labels))       
+                ($exemplar-fts-size (size-atom $exemplar-fts)) 
+                ($init-ft-indices (map-atom $init-fts $ft (List.index $labels $ft)))                  ;; initial ft names changed into index values
+
+                ($ft-set-pop (if $init-xmplr-fts                                                      ;; if the featue selection is to be initialized with exemplar features
+                                (let* (($updated-init-fts (union-atom $init-fts $exemplar-fts))       ;; include exemplar featues in the init features list(tuple)
+                                        ($updated-target-size (+ $target-size $exemplar-fts-size))    ;; target size also should be updated
+                                        ($updated-init-fts (union-atom $init-ft-indices $exemplar-fts))) 
```

Comment:
The variable `$update-init-fts` has duplicate definition both here and on line 31.